Lib/test/test_pyrepl/test_reader.py

```diff
@@ -355,3 +362,60 @@ def test_setpos_from_xy_for_non_printing_char(self):
         reader, _ = handle_all_events(events)
         reader.setpos_from_xy(8, 0)
         self.assertEqual(reader.pos, 7)
+
+    def test_syntax_highlighting_basic(self):
```

Comment:
I would recommend adding some extra test for edge cases and failure modes such as incomplete code blocks, syntax errors and other known failure conditions so we know how this will behave